src/node_api_jsrt.cc

```diff
@@ -221,43 +265,100 @@ namespace jsrtimpl {
     }
 
   private:
+    napi_env _env;
     napi_callback _cb;
     void* _data;
   };
 
-}  // end of namespace jsrtimpl
+  inline napi_status JsPropertyIdFromKey(JsValueRef key,
+                                         JsPropertyIdRef* propertyId) {
+    JsValueType keyType;
+    CHECK_JSRT(JsGetValueType(key, &keyType));
 
-#define RETURN_STATUS_IF_FALSE(condition, status)                       \
-  do {                                                                  \
-    if (!(condition)) {                                                 \
-      return napi_set_last_error((status));                             \
-    }                                                                   \
-  } while(0)
+    if (keyType == JsString) {
+      size_t length;
+      CHECK_JSRT_EXPECTED(
+        JsCopyStringUtf8(key, nullptr, 0, &length), napi_string_expected);
 
-#define CHECK_ARG(arg)                                                  \
-  RETURN_STATUS_IF_FALSE((arg), napi_invalid_arg)
+      std::vector<uint8_t> name;
+      name.reserve(length + 1);
+      CHECK_JSRT(JsCopyStringUtf8(key, name.data(), length + 1, &length));
 
-#define CHECK_JSRT(expr)                                                \
-  do {                                                                  \
-    JsErrorCode err = (expr);                                           \
-    if (err != JsNoError) return napi_set_last_error(err);              \
-  } while(0)
+      CHECK_JSRT(JsCreatePropertyIdUtf8(
+        reinterpret_cast<char*>(name.data()), length, propertyId));
+    } else {
+      CHECK_JSRT(JsGetPropertyIdFromSymbol(key, propertyId));
```

Comment:
Should we not check explicitly here for keyType being a symbol? Should JSRT expose a symbol keytype to make this check easier here? Or are you just relying on JsGetPropertyIdFromSymbol to fail if anything other than a symbol is passed in?